File: user-service/src/main/java/com/user/service/exceptions/exceptions/user/found/UserNotFoundException.java
package com.user.service.exceptions.exceptions.user.found;

import com.user.service.exceptions.exceptions.user.main.UserException;

@SuppressWarnings("unused")
public class UserNotFoundException extends UserException {

    public static UserNotFoundException create(Long id) {
        return new UserNotFoundException("User not found by id: " + id);
    }

    public UserNotFoundException() {
    }

    public UserNotFoundException(String message) {
        super(message);
    }

    public UserNotFoundException(String message, Throwable cause) {
        super(message, cause);
    }

    public UserNotFoundException(Throwable cause) {
        super(cause);
    }

    public UserNotFoundException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
        super(message, cause, enableSuppression, writableStackTrace);
    }
}

File: notification-service/src/main/kotlin/ua/notification/service/exception/request/found/UserNotFoundException.kt
package ua.notification.service.exception.request.found

import ua.notification.service.exception.request.main.DirectNotificationException

@Suppress("unused")
class UserNotFoundException : DirectNotificationException {
    constructor() : super()
    constructor(message: String?) : super(message)
    constructor(message: String?, cause: Throwable?) : super(message, cause)
    constructor(cause: Throwable?) : super(cause)
    constructor(message: String?, cause: Throwable?, enableSuppression: Boolean, writableStackTrace: Boolean) : super(
        message,
        cause,
        enableSuppression,
        writableStackTrace
    )

    companion object {
        fun create(id: Long): UserNotFoundException {
            return UserNotFoundException("User not found by id: $id")
        }
    }
}